tools/tsp-client/src/commands.ts

```diff
@@ -480,13 +480,49 @@ export async function generateConfigFilesCommand(argv: any) {
   if (Object.keys(overrideJson).length > 0) {
     emitterPackageJson["overrides"] = overrideJson;
   }
-  
-  const emitterPath = resolveEmitterPathFromArgs(argv) ?? joinPaths(await getRepoRoot(outputDir), defaultRelativeEmitterPackageJsonPath);
-  
-  await writeFile(
-    emitterPath,
-    JSON.stringify(emitterPackageJson, null, 2),
-  );
+
+  const emitterPath =
+    resolveEmitterPathFromArgs(argv) ??
+    joinPaths(await getRepoRoot(outputDir), defaultRelativeEmitterPackageJsonPath);
+
+  try {
+    const existingEmitterPackageJson = JSON.parse(await readFile(emitterPath, "utf8"));
+    // If there's an existing emitter-package.json, we need to merge any untracked devDependencies
+    if (existingEmitterPackageJson) {
+      const currentDevDependencies = existingEmitterPackageJson["devDependencies"] ?? {};
+      const packageLockPath = joinPaths(dirname(packageJsonPath), "package-lock.json");
+      let existingPackageLockJson;
+      try {
+        // Attempt to read package-lock.json
+        existingPackageLockJson = JSON.parse(await readFile(packageLockPath, "utf8"));
+      } catch (err) {
+        Logger.debug(`Unable to read package-lock.json: ${packageLockPath}`);
+      }
+      // If there are untracked dependencies, we will check the package-lock.json of the
+      // target emitter for a compatible version, otherwise we will leave the version from
+      // the emitter-package.json unchanged
+      for (const key of Object.keys(currentDevDependencies)) {
+        if (
+          emitterPackageJson["devDependencies"] &&
```

Comment:
What if there is no "devDependencies" in the "current" not yet written version? Don't we still need to write any dev dependencies from the existing version?

Reply:
But I thought this loop is trying to retain the manually added dependencies from the emitter-package-json. I'm wondering why we wouldn't have the same behavior if the generated emitter package json happens to not have a devDependencies section. i.e. why is this special?

Reply:
Ah I think I misunderstood your first question, you're talking about the case where we dont have any pinned dev deps. I didnt take it into account since I dont think we have any emitter that doesnt, but that doesnt mean someone wont run into this. I'll support that corner case, thanks for the feedback!